lambdas/shared/t4_lambda_shared/decorator.py

```diff
@@ -0,0 +1,58 @@
+"""
+Decorators for using lambdas in API Gateway
+"""
+
+from functools import wraps
+import traceback
+
+from jsonschema import Draft4Validator, ValidationError
+
+def api(cors_origins=[]):
+    def innerdec(f):
+        @wraps(f)
+        def wrapper(event, _):
+            params = event['queryStringParameters'] or {}
+            headers = event['headers'] or {}
+            try:
+                status, body, response_headers = f(params, headers)
+            except Exception:
```

Comment:
this deserves a docstring mention: we will literally catch anything, e.g. KeyboardInterrupt (which shouldn't be a factor in lambda functions)

Reply:
Nope - `KeyboardInterrupt` is not an `Exception`